### PR TITLE
Template Selection plugin

### PR DESCRIPTION
**Pull request by yann300**
https://github.com/ethereum/remix-project/issues/4482

### CODE REVIEW

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
+
+import React from 'react'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { CustomTooltip } from "@remix-ui/helper";
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:4]: semicolon can be removed

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+  renderComponent() {
+    this.dispatch({
+      ...this,
+    })
+  }
+
+  updateComponent() {
+    /*
+    const opts = {
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:71]: remove commented code

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      this.emit('createWorkspaceReducerEvent', workspaceName, item.value, item.opts, false, (e, data) => {
+        if (e) {
+          const modal: AppModal = {
+            id: 'TemplatesSelection',
+            title:  window._intl.formatMessage({ id: !isElectron() ? 'filePanel.workspace.create': 'filePanel.workspace.create.desktop' }),
+            message: e.message,
+            okLabel: window._intl.formatMessage({ id: 'filePanel.ok' }),
+            cancelLabel: window._intl.formatMessage({ id: 'filePanel.cancel' })
+          }
+          this.call('notification', 'modal', modal)
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:107]: `await` should be added

File: apps/remix-ide/src/app/plugins/templates-selection/templates.ts [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+      ]
+    },
+    {
+      name: "Contract Verification",
+      items: [
+        { value: "etherscanScripts", displayName: intl.formatMessage({ id: 'filePanel.addscriptetherscan' }) },
+      ],
+    },
+    {
+      name: 'Github Actions',
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates.ts:279]: it should be `GitHub`

File: libs/remix-ui/workspace/src/lib/actions/workspace.ts [hunk 5]
```diff
@@ -150,6 +151,7 @@ export const createWorkspace = async (
     await plugin.workspaceCreated(workspaceName)
 
     if (isGitRepo && createCommit) {
+      console.log('CREATE COMMIT')
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/actions/workspace.ts:154]: remove log

**Review by Aniket-Engg — COMMENTED** [review]
I get this error while loading remix using netlify link

Screenshot 2024-06-18 at 12 21 51 PM

unchecking the ERC20 box from tags ruins the symmetry in UI
Screenshot 2024-06-18 at 12 12 29 PM

**Comment by Aniket-Engg** [comment]
should there be `Add to current` option for `blank` template?

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        showPin={false}
+        tagList={[
+          ['Solidity', 'danger'],
+          ['ZKP', 'warning'],
+          ['ERC20', 'success'],
+          ['ERC721', 'secondary'],
+          ['ERC1155', 'primary'],
+        ]}
+        title='Template explorer'
+        description="Select the template to create a workspace"
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:147]: it should also say `"Select the template to create a workspace or to add in current workspace"`

**Comment by Aniket-Engg** [comment]
It is difficult to identify the different between multiple templates with same details like `ERC721(NFT)` or `ERC20`

**Comment by Aniket-Engg** [comment]
not sure if `Contract Verification`, `Solidity CREATE2`, `Generic ZKP` should have `Create` option??

**Comment by Aniket-Engg** [comment]
Screenshot 2024-06-18 at 12 35 36 PM

plugin is displayed in plugin list and saying `Not maintained by Remix`

**Comment by Aniket-Engg** [comment]
Screenshot 2024-06-18 at 12 35 36 PM

plugin is displayed in plugin list and saying `Not maintained by Remix`

File: apps/remix-ide/src/app/plugins/templates-selection/templates.ts [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+      items: [
+        { value: "runJsTestAction", displayName: intl.formatMessage({ id: 'filePanel.tssoltestghaction' }) },
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates.ts:281]: There is same name and description added for js test workflow and solidity test workflow. Should be updated

**Comment by Aniket-Engg** [comment]
Clicking on `Add to current` takes time some time but user does not get any info about what's happening

**Comment by Aniket-Engg** [comment]
Screenshot 2024-06-18 at 1 02 56 PM

typing anything in `Filter` textbox shows lots of logs in browser console

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+        showUntagged={true}
+        showPin={false}
+        tagList={[
+          ['Solidity', 'danger'],
+          ['ZKP', 'warning'],
+          ['ERC20', 'success'],
+          ['ERC721', 'secondary'],
+          ['ERC1155', 'primary'],
+        ]}
+        title='Template explorer'
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:146]: not sure if it should be named as `Workspace Template Explorer`

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx (hunk 3 above)
Comment (yann300) [reply apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:107]: it's not really necessary here I think.

File: apps/remix-ide/src/app/plugins/templates-selection/templates.ts (hunk 4 above)
Comment (yann300) [reply apps/remix-ide/src/app/plugins/templates-selection/templates.ts:279]: this section is really specific about github actions.

**Comment by yann300** [comment]
> not sure if `Contract Verification`, `Solidity CREATE2`, `Generic ZKP` should have `Create` option??

I think that's fine..

File: apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx (hunk 8 above)
Comment (yann300) [reply apps/remix-ide/src/app/plugins/templates-selection/templates-selection-plugin.tsx:146]: I would say `Template explorer'` is ok

**Comment by Aniket-Engg** [comment]
Screenshot 2024-07-16 at 1 48 37 PM
 UI seems broken.

**Comment by LianaHus** [comment]
> Screenshot 2024-07-16 at 1 48 37 PM UI seems broken.

yes because I changed a bit the grid view in the other PR. I'll fix now

**Comment by Aniket-Engg** [comment]
Screenshot 2024-07-17 at 12 10 40 PM

UI still seems broken to me

**Comment by Aniket-Engg** [comment]
Screenshot 2024-07-18 at 7 46 01 PM

I think text should be only `Browse` with some difference in button style like we usually see on different platforms

**Comment by yann300** [comment]
This PR still needs:
 - The search feature to be checked.
 - The pin icon to be improved (with a border when it's checked and so on..)

**Comment by LianaHus** [comment]
> Screenshot 2024-07-18 at 7 46 01 PM
> I think text should be only `Browse` with some difference in button style like we usually see on different platforms

this should be removed

File: apps/remix-ide/src/app/plugins/templates-selection/templates.ts [hunk 9]
```diff
@@ -49,13 +51,15 @@ export const templates = (intl) => {
...
           tagList: ["Solidity"],
+          description: "A standard interface for contracts that manage multiple token types",
           opts: {
             mintable: true
           }
         },
         {
           value: "ozerc20",
           displayName: "ERC20",
+          description: "A standard interface for fungible tokens",
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/templates-selection/templates.ts:62]: to distinguish it with other one, we can make it like: `A standard interface for mintable fungible tokens` and similarly

Comment (LianaHus) [reply apps/remix-ide/src/app/plugins/templates-selection/templates.ts:62]: we have tags there, no need to repeat it in description

**Comment by Aniket-Engg** [comment]
Screenshot 2024-07-23 at 6 40 07 PM

UI still breaks for 1155 template with tags

**Review by joeizang — DISMISSED** [review]
There are multiple errors in the console that should be fixed. See the image included in this comment:
image

Also the buttons don't seem to be off like @Aniket-Engg mentioned.

**Review by Aniket-Engg — APPROVED** [review]
Functionality wise looks fine to me

**Pull request merged**